### PR TITLE
[UNR-2689][MS] Spatial output log will pop up at the end of PIE sessi…

### PR DESCRIPTION
**Pull request by MatthewSandfordImprobable**
…on. A setting can be changed to disable this.

**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
Spatial output log will now be displayed on PIE session end. This is similar to message output functionality. There is also a setting to stop the output popping up. If the spatial output is already open, the window will not be modified or pop up again annoyingly.  

#### Release note
Changelog updated

QA Testing:
Open up example project. Ensure that the Spatial output log is not turned on. Start and stop a PIE session. Note that the Spatial output log is now visible. Now close the Spatial output log and disable the "OpenSpatialOutputLogOnPIESessionEnd" inside "Service Settings". Test again. The Spatial output log should not pop up.

#### Primary reviewers
@jessicafalk  @joshuahuburn

### CODE REVIEW

**Comment by MatthewSandfordImprobable** [comment]
Also look at: 
https://github.com/improbableio/UnrealEngine/compare/4.23-SpatialOSUnrealGDK...improbableio:feature/UNR-2689?expand=1

File: SpatialGDK/Source/SpatialGDKServices/Public/SpatialGDKServiceSettings.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
```
Comment (jessicafalk) [inline SpatialGDK/Source/SpatialGDKServices/Public/SpatialGDKServiceSettings.h]: nice! this should make it much easier to have some settings that can be used by all modules

File: CHANGELOG.md [hunk 2]
```diff
@@ -50,6 +50,7 @@ Usage: `DeploymentLauncher createsim <project-name> <assembly-name> <target-depl
 - Moved Dev Auth settings from runtime settings to editor settings.
 - Added the option to use the development authentication flow using the command line.
 - Added a button to generate the Development Authentication Token inside the Unreal Editor. To use it, navigate to **SpatialOS GDK for Unreal** > **Editor Settings** > **Cloud Connection**.
+- The Spatial output log will not automatically pop up at the end of a PIE session if it is not already open. This is to make spatial errors more obvious to customers. This functionality can be stopped by disabling OpenSpatialOutputLogOnPIESessionEnd in **Service Settings**.
```
Comment (jessicafalk) [inline CHANGELOG.md]:
```suggestion
- The Spatial output log will now automatically pop up at the end of a PIE session if it is not already open. This is to make spatial errors more obvious to customers. This functionality can be stopped by disabling OpenSpatialOutputLogOnPIESessionEnd in **Service Settings**. 
```
?

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorModule.cpp [hunk 3]
```diff
@@ -73,6 +84,14 @@ void FSpatialGDKEditorModule::UnregisterSettings()
 	PropertyModule.UnregisterCustomPropertyTypeLayout("FWorkerAssociation");
 }
 
+bool FSpatialGDKEditorModule::HandleServiceSettingsSaved()
+{
+	GetMutableDefault<USpatialGDKServiceSettings>()->SaveConfig();
+
```
Comment (jessicafalk) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorModule.cpp]: nit: probably not need that new line here

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorModule.cpp [hunk 4]
```diff
@@ -73,6 +84,14 @@ void FSpatialGDKEditorModule::UnregisterSettings()
...
+	return true;
+}
+
+
```
Comment (jessicafalk) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorModule.cpp]: nit: new line

**Comment by MatthewSandfordImprobable** [comment]
This PR has changed quite a lot. The changes are now very small but fulfill the Jira ticket more accurately.

**Comment by jessicafalk** [comment]
this PR doesn't seem to have any changes now except for the changelog :)

**Comment by MatthewSandfordImprobable** [comment]
Indeed, the only changes are in the engine (and changelog).

**Pull request merged**